File: TestP1/src/irrgarten/TestP1.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template
 */
package irrgarten;

import java.util.ArrayList;
import interfaz.GuiUI;

/**
 *
 * @author zapi24
 */
public class TestP1 {

    //CREAMOS LOS DISTINTOS ENUMERADOS UTILIZADOS EN LAS DISTINTAS CLASES DEL PROYECTO
    
    public static void main(String[] args) {
        // TODO code application logic here
       
        TextUI vista_texto=new TextUI();
        
        
        Game juego=new Game(1,true); //TRUE para modo debug activado
        Controller controlador=new Controller(juego,vista_texto);
        
        
        boolean mode=false; //Para indicar el modo de vista
        
        if(mode){
            
            GuiUI pantalla = new GuiUI();
            VController vcontrolador = new VController(juego,pantalla);
            vcontrolador.play();
        }
        else{
            
            controlador.play(); //Modo de texto
        }
        
        
        
        //TEST DE LOS ENUM
        
       /* Directions abajo=Directions.DOWN;
        
        System.out.print(abajo+"\n"+"\n");
        
        //TEST DE CLASE WEAPON
        
        Weapon armatest=new Weapon(2,4);
        System.out.print(armatest.discard()+"\n");
        System.out.print(armatest.toString()+"\n"+"\n");
        
        //TEST DE CLASE SHIELD
        
       Shield escudotest=new Shield(3,4);
       System.out.print(escudotest.discard()+"\n");
       System.out.print(escudotest.toString()+"\n"+"\n");
       
       //TEST DE LA CLASE DICE
       
       //Dice dicetest=new Dice();
       System.out.print(Dice.randomPos(10)+"\n");
       System.out.print(Dice.whoStars(5)+"\n");
       System.out.print(Dice.randomIntelligence()+"\n");
       System.out.print(Dice.randomStrength()+"\n"+"\n");
       
       System.out.print(Dice.resurrectPlayer()+"\n"+"\n");
       
       System.out.print(Dice.weaponsReward()+"\n");
       System.out.print(Dice.shieldsReward()+"\n");
       System.out.print(Dice.healthReward()+"\n");
       System.out.print(Dice.weaponPower()+"\n");
       System.out.print(Dice.shieldPower()+"\n");
       System.out.print(Dice.usesLeft()+"\n"+"\n");
       
       System.out.print(Dice.discardElement(3)+"\n"+"\n");
       
       //TEST DE LA CLASE MONSTER
       
       Monster monstruo=new Monster("Jhonny",10,10);
       
       System.out.print(monstruo.dead()+"\n");
       System.out.print(monstruo.attack()+"\n");
       
       monstruo.setPos(2, 3);
       System.out.print(monstruo.toString()+"\n"+"\n");
       monstruo.gotWounded();
       
       //TEST DE LA CLASE JUGADOR
       
       Player jugador=new Player('0',10,10);
       jugador.setPos(2, 3);
       System.out.print(jugador.toString()+"\n");
       
       jugador.gotWounded();
       System.out.print(jugador.toString()+"\n"+"\n");
       
       //TEST DE LA CLASE LABYRINTH
       
       Labyrinth laberinto=new Labyrinth(5,7,4,4);
       System.out.print(laberinto.toString()+"\n");
       System.out.print(laberinto.haveAWinner()+"\n");
       
       laberinto.addMonster(2, 3, monstruo);
       System.out.print(laberinto.toString()+"\n");
       
       System.out.print(laberinto.posOK(9, 0)+"\n");
       System.out.print(laberinto.emptyPos(2, 2)+"\n");
       System.out.print(laberinto.monsterPos(2, 2)+"\n");
       System.out.print(laberinto.exitPos(4, 4)+"\n");
       
       int posicion[]=new int[2];
       
       posicion=laberinto.dir2Pos(3, 3, Directions.LEFT);
       System.out.print("("+ posicion[0]+","+posicion[1]+")"+"\n"); 
       
       
       posicion=laberinto.randomEmptyPos();
       System.out.print("("+ posicion[0]+","+posicion[1]+")"+"\n"+"\n");  //(2,3) es la unica no vacia, hay un monstruo
       
       
       
       //TEST DE LA CLASE GAME
       
       Game juego=new Game(5);
       System.out.print(juego.finished()+"\n");
       System.out.print(juego.getLaberinto().toString()+"\n");
       
       juego.logMonsterWon();
       juego.logPlayerWon();
       juego.nextPlayer();
       juego.logResurrected();
       
       //Test de la clase GameState, partiendo de Game
       GameState juego_actual= juego.getGameState();
       
       System.out.print(juego_actual.getPlayers());
       System.out.print(juego_actual.getMonsters());
       System.out.print(juego_actual.getLog()+"\n");
       System.out.print(juego_actual.getCurrentPlayer()+"\n");
       
       
      
       System.out.print(juego.getLog()+"\n");*/
    }
    
}

File: Irrgarten_ruby/testp1.rb
# frozen_string_literal: true

module Irrgarten

  require_relative 'irrgarten'
  require_relative 'shield.rb'
  require_relative 'weapon.rb'
  require_relative 'dice.rb'
  require_relative 'game_state'
  require_relative 'monster'
  require_relative 'player'
  require_relative 'labyrinth'
  require_relative 'game'
  require_relative 'controller'
  require_relative 'text_ui'
  d


  class Testp1 #EQUIVALENTE A  MAIN

    vista = TextUI.new
    juego = Game.new_game(1,true) #True para modo debug activado
    controlador = Controller.new(juego,vista)

    controlador.play


    # #PRUEBAS DE DIRECTIONS
    #
    #     puts Directions::RIGHT
    #     puts Directions::LEFT
    #     puts
    #     #PRUEBAS DE ESCUDO
    #
    #     escudo= Shield.new(3.0,4)
    #
    #     puts escudo.to_s
    #     escudo.protect
    #     puts escudo.to_s
    #     puts
    #     #PRUEBAS DE WEAPON
    #
    #     arma= Weapon.new(2.0,4)
    #
    #     puts arma.to_s
    #     arma.attack
    #     puts arma.to_s
    #     puts
    #     #PRUEBAS DE DICE
    #
    #     puts Dice.random_pos(5)
    #
    #     #PRUEBAS DE GAME STATE
    #     juego= Game_state.new("laberinto","jugadores","monstruos",0,true,"logg")
    #
    #     puts juego.winner
    #     puts juego.monsters
    #     puts juego.labyrinthv
    #     puts
    #
    #     #PRUEBAS DE MONSTER
    #
    #     monstruo=Monster.new_monster("MonstruoEj",10,10)
    #
    #     puts monstruo.dead
    #     puts monstruo.attack
    #
    #     monstruo.set_pos(2,2)
    #     puts monstruo.to_s
    #     puts monstruo.got_wounded
    #     puts
    #
    #     #PRUEBAS DE PLAYER
    #
    #     jugador=Player.new_player(0,10,10)
    #
    #     puts jugador.dead
    #     jugador.set_pos(2,3)
    #     puts jugador.to_s
    #
    #     arma=jugador.new_weapon
    #     escudo=jugador.new_weapon
    #     puts arma.to_s
    #     puts escudo.to_s
    #     puts
    #
    #     #PRUEBAS DE LABYRINTH
    #     laberinto=Labyrinth.new_labyrinth(5,7,4,4)
    #     puts laberinto.have_a_winner
    #     puts laberinto.to_s
    #
    #     laberinto.add_monster(2,3,monstruo)
    #     puts laberinto.to_s
    #
    #     puts laberinto.pos_ok(9,0)
    #     puts laberinto.empty_pos(2,2)
    #     puts laberinto.monster_pos(2,2)
    #     puts laberinto.exit_pos(4,4)
    #
    #     #posicon=Array.new(2)
    #     posicion=laberinto.dir_2pos(3,3,Directions::LEFT)
    #     puts "(#{posicion[0]},#{posicion[1]})"
    #
    #     posicion=laberinto.random_empty_pos
    #     puts "(#{posicion[0]},#{posicion[1]})"
    #     puts
    #
    #     #PRUEBAS DE CLASE GAME
    #     juego = Game.new_game(5,true)
    #     puts juego.finished
    #     puts juego.laberinto.to_s
    #
    #     juego.log_monster_won
    #     juego.log_player_won
    #     juego.next_player
    #     juego.log_resurrected
    #
    #     juego_actual = juego.get_game_state
    #
    #     puts juego_actual.players
    #     puts juego_actual.monsters
    #     puts juego_actual.log
    #     puts juego_actual.current_player


  end
end



